src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -345,15 +345,18 @@ public void deleteElement(UUID elementUuid, String userId) {
         );
     }
 
-    private void deleteObject(ElementAttributes elementAttributes) {
+    private void deleteObject(ElementAttributes elementAttributes, String userId) {
         if (elementAttributes.getType().equals(DIRECTORY)) {
-            deleteSubElements(elementAttributes.getElementUuid());
+            deleteSubElements(elementAttributes.getElementUuid(), userId);
         }
         directoryElementRepository.deleteById(elementAttributes.getElementUuid());
+        if (STUDY.equals(elementAttributes.getType())) {
```

Comment:
Ok to emit notification only if the element is of study type for now, but we could have a generic notification for element deletion

Reply:
that was the goal indeed, I handled the study deletion and nothing else. The day we need more generic notification it won't be hard to change.